src/if_python3.c

```diff
@@ -860,12 +860,25 @@ Python3_Init(void)
 
 	init_structs();
 
-
-#ifdef PYTHON3_HOME
-# ifdef DYNAMIC_PYTHON3
-	if (mch_getenv((char_u *)"PYTHONHOME") == NULL)
-# endif
+#ifdef DYNAMIC_PYTHON3
+	if (p_py3home && *p_py3home != '\0')
+	{
+	    int len;
+	    wchar_t *buf;
+	    len = strlen((char *)p_py3home);
```

Comment:
I think it is better to calculate `len` by `mbstowcs(NULL, (char *)p_py3home, 0)`, and it needs `+ 1` for wide-nul character.

```c
len = mbstowcs(NULL, (char *)p_py3home, 0);
buf = (wchar_t *)alloc((len + 1) * sizeof(wchar_t));
if (buf && mbstowcs(buf, (char *)p_py3home, len + 1) != (size_t)-1)
    Py_SetPythonHome(buf);
```